x-pack/plugins/security_solution/public/detections/components/rules/step_about_rule/schema.tsx

```diff
@@ -38,6 +39,19 @@ export const schema: FormSchema<AboutStepRule> = {
       }
     ),
     labelAppend: OptionalFieldLabel,
+    validations: [
+      {
+        validator: emptyArrayItem(
```

Comment:
I think that use can use `emptyField` instead of the custom `emptyArrayItem` function.

Reply:
That would be a way cleaner solution, but I wasn't able to make it work. The problem that I had was that  `emptyField` not only validates an empty string but also an empty array. Here is the code:
```
export const emptyField = (message: string) => (
  ...args: Parameters<ValidationFunc>
): ReturnType<ValidationFunc<any, ERROR_CODE>> => {
  const [{ value, path }] = args;

  if (typeof value === 'string') {
    return isEmptyString(value) ? { code: 'ERR_FIELD_MISSING', path, message } : undefined;
  }

  if (Array.isArray(value)) {
    return isEmptyArray(value) ? { code: 'ERR_FIELD_MISSING', path, message } : undefined;
  }
};
```

The extra array validation makes the field mandatory. But author and tags shouldn't be required fields. 
<img width="954" alt="Screenshot 2021-06-14 at 15 55 37" src="https://user-images.githubusercontent.com/1490444/121905411-6a429100-cd2a-11eb-938d-0a563602d4a3.png">


Do you know a way to avoid this problem?

Reply:
I think you can use the `isBlocking` property (`src/plugins/es_ui_shared/static/forms/hook_form_lib/types.ts`). For example:

```
validations: [
      {
        validator: emptyField(...),
        isBlocking: false,
      },
    ],
```

If this doesn't work then we can leave the `emptyArrayItem` as it is and move forward.

Reply:
### Context

We don't want users to create empty items on a COMBO_BOX, but we also want to keep this field optional. To do that, I made a new validator called `emptyArrayItem `.

@cnasikas suggested a  much cleaner alternative that doesn't require the creation of a new validator.  
```js
{
  validator: emptyField(...),
  isBlocking: false,
  type: 'arrayItem',
}
```

It does the work, but it has an unexpected side effect. When the user tries to go to the next step, the validation blocks the user from proceeding. But when they try for the second time, it works (video attached). 

https://user-images.githubusercontent.com/1490444/122196782-cf67c500-ce97-11eb-943b-84b465a6dfff.mp4


### Question
 @elastic/kibana-core Is it possible to use `emptyField` for this scenario? Are we missing something trivial here?

Thank you for your attention!

Reply:
I'd need to see the code but @cnasikas is right that it should work with the `isBlocking: false` on the validator.

Reply:
> @elastic/kibana-core Is it possible to use emptyField for this scenario?

Sorry, what's that? Are we talking about that?

https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/src/plugins/es_ui_shared/static/forms/helpers/field_validators/empty_field.ts#L14-L26

In that case, the owner seems to be @elastic/kibana-stack-management